WalletWasabi.Gui/ViewModels/StatusBarViewModel.cs

```diff
@@ -253,6 +257,27 @@ public void Initialize(NodesCollection nodes, WasabiSynchronizer synchronizer)
 
 					UpdateAvailable = !x.ClientUpToDate;
 					CriticalUpdateAvailable = !x.BackendCompatible;
+
+					// No locking should be fine as it's happening on the same thread.
+					if (InLegalTrouble)
+					{
+						return;
+					}
+
+					try
+					{
+						InLegalTrouble = true;
+						if (Global.LegalDocuments is null || Global.LegalDocuments.Version != x.LegalDocumentsVersion)
+						{
+							var legalResp = await LegalDocuments.FetchLatestAsync(Global.DataDir, () => Global.Config.GetCurrentBackendUri(), Global.Config.TorSocks5EndPoint, CancellationToken.None);
```

Comment:
I would not add business logit into a viewmodel. Can you move this into UpdateChecker?

Reply:
Not sure it's business logic. The logic here decides if a UI tab should be opened or not.

Reply:
But if you have recommendation, I'd be happy to apply it if it makes sense.

Reply:
Removed InLegalTrouble as you mentioned because we using a single thread, no way of calling this twice.